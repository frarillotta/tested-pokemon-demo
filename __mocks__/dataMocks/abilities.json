[
	{
		"ability":{
		"name":"torrent",
		"url":"https://pokeapi.co/api/v2/ability/67/"
		},
		"is_hidden":false,
		"slot":1
	},
	{
		"ability":{
		"name":"rain-dish",
		"url":"https://pokeapi.co/api/v2/ability/44/"
		},
		"is_hidden":true,
		"slot":3
	}
]