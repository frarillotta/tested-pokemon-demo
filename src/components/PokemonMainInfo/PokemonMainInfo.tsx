import styled from "styled-components";
import { getHeight, getWeight, capitalizeFirstLetter } from "../../utils";

const PokemonMainInfo = ({height, weight, abilities}) => {
	return <PokemonInfo>
		<InfoPropertyValueWrapper>
			<InfoProperty>
				Height:
			</InfoProperty>
			<InfoValue>
				{getHeight(height)}
			</InfoValue>
		</InfoPropertyValueWrapper>
		<InfoPropertyValueWrapper>
			<InfoProperty>
				Weight: 
			</InfoProperty>
			<InfoValue>
				{getWeight(weight)}
			</InfoValue>
		</InfoPropertyValueWrapper>
		<InfoPropertyValueWrapper>
			<InfoProperty>
				Abilities: 
			</InfoProperty>
			<InfoValue>
				{abilities.reduce((accumulator, currentVal, index) => {
					const formattedAbility = capitalizeFirstLetter(currentVal.ability.name).replace("-", " ");
					const val = accumulator + (index === 0 ? "" : ", ") + formattedAbility;
					return val
				}, "")}
			</InfoValue>
		</InfoPropertyValueWrapper>
	</PokemonInfo>
}


const InfoPropertyValueWrapper = styled.div`
	display: grid;
	grid-template-areas: 
		"propertyname propertyvalue propertyvalue .";
	grid-template-columns: repeat(3, 1fr);
	align-items: center;
	gap: 2rem;
`

const InfoProperty = styled.span`
	grid-area: propertyname;
	color: #878787
`

const InfoValue = styled.span`
	grid-area: propertyvalue;
`

const PokemonInfo = styled.div`
	display: grid;
	gap: 20px;
	padding: 25px;
`

export { PokemonMainInfo }