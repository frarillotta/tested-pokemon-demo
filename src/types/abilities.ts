import type { URLString } from './pokemon';

export type Abilities = {
    effect_changes: [],
    effect_entries: {
          effect: string,
          language:{
             name: string,
             url:URLString
          },
          short_effect:string
    }[],
    flavor_text_entries: {
          flavor_text: string,
          language:{
             name: string,
             url: URLString
          },
          version_group:{
            name: string,
            url: URLString
          }
    }[],
    generation:{
       name: string,
       url: URLString
    },
    id: number,
    is_main_series: boolean,
    name: string,
    names:{
          language:{
            name: string,
            url: URLString
          },
          name: string
       }[],
    pokemon: {
          is_hidden: boolean,
          pokemon:{
            name: string,
            url: URLString
          },
          slot: number
    }[]
 }