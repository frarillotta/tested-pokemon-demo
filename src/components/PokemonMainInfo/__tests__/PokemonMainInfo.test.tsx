import { render } from "@testing-library/react";
import {PokemonMainInfo} from "../PokemonMainInfo";
import abilities from "../../../../__mocks__/dataMocks/abilities.json";

describe("PokemonMainInfo", () => {
	
	describe("when loaded", () => {
		
		it("should have a consistent snapshot", () => {
			const {container} = render(<PokemonMainInfo abilities={abilities} height={90} weight={90} />);
			expect(container).toMatchSnapshot();
		});

	});

})