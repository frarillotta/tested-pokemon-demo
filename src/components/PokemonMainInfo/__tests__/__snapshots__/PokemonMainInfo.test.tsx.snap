// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonMainInfo when loaded should have a consistent snapshot 1`] = `
<div>
  <div
    class="PokemonMainInfo__PokemonInfo-sc-1y1zaw-3 eFwIDo"
  >
    <div
      class="PokemonMainInfo__InfoPropertyValueWrapper-sc-1y1zaw-0 jVYAZk"
    >
      <span
        class="PokemonMainInfo__InfoProperty-sc-1y1zaw-1 gtLCOd"
      >
        Height:
      </span>
      <span
        class="PokemonMainInfo__InfoValue-sc-1y1zaw-2 hqHXIZ"
      >
        900 cm ~(29'6'')
      </span>
    </div>
    <div
      class="PokemonMainInfo__InfoPropertyValueWrapper-sc-1y1zaw-0 jVYAZk"
    >
      <span
        class="PokemonMainInfo__InfoProperty-sc-1y1zaw-1 gtLCOd"
      >
        Weight:
      </span>
      <span
        class="PokemonMainInfo__InfoValue-sc-1y1zaw-2 hqHXIZ"
      >
        9 kg (19.85 lbs)
      </span>
    </div>
    <div
      class="PokemonMainInfo__InfoPropertyValueWrapper-sc-1y1zaw-0 jVYAZk"
    >
      <span
        class="PokemonMainInfo__InfoProperty-sc-1y1zaw-1 gtLCOd"
      >
        Abilities:
      </span>
      <span
        class="PokemonMainInfo__InfoValue-sc-1y1zaw-2 hqHXIZ"
      >
        Torrent, Rain dish
      </span>
    </div>
  </div>
</div>
`;
